### PR TITLE
Library Framework Update

### PR DESCRIPTION
**Pull request by eslutz**
## Description

Updated target frameworks to net6.0, netcoreapp3.1, and net48.

Added .editorconfig and Directory.Build.props for consistency.

## Type of change: 4

1. Non-functional change (e.g. documentation changes, removing unused `using` directives, renaming local variables, etc)
2. Bug fix (non-breaking change that fixes an issue)
3. New feature (non-breaking change that adds functionality)
4. Breaking change (fix or feature that could cause existing functionality to not work as expected)
---

<sup>_[Reviewer guidelines](../blob/main/CONTRIBUTING.md#reviewing-changes)_</sup>

### CODE REVIEW

File: RockLib.Diagnostics/Tracing.cs [hunk 1]
```diff
@@ -13,7 +13,10 @@ namespace RockLib.Diagnostics
     /// Defines methods for configuring the <see cref="Trace"/> static class and retrieving
     /// <see cref="TraceSource"/> objects.
     /// </summary>
+#pragma warning disable CA1724
```
Comment (eslutz) [inline RockLib.Diagnostics/Tracing.cs:16]: [CA1724: Type names should not match namespaces](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1724) - Suggested to suppress for already existing libraries.

File: RockLib.Diagnostics.ConfigTests/Tracing/CustomTraceListener.cs [hunk 2]
```diff
@@ -1,12 +1,15 @@
 ﻿using System;
 using System.Diagnostics;
+
+#pragma warning disable CA1050 // Declare types in namespaces
```
Comment (eslutz) [inline RockLib.Diagnostics.ConfigTests/Tracing/CustomTraceListener.cs:4]: [CA1050: Declare types in namespaces](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1050) - It says "It is safe to suppress a warning from this rule if performance is not a concern" so I'm suppressing it after spending time unsuccessfully adding it to a namespace without causing test failures.

File: RockLib.Diagnostics.Tests/ConfigurationExtensions/CreateDiagnosticsSettingsTests.cs [hunk 3]
```diff
@@ -1,159 +1,164 @@
...
-                { "sources:switch:level", "Error" },
-                { "sources:listeners:type", typeof(DefaultTraceListener).AssemblyQualifiedName },
-                { "sources:listeners:value:name", "listener2" },
-                { "sources:listeners:value:logFileName", "listener2.log" },
-                { "sources:listeners:value:filter:type", "System.Diagnostics.EventTypeFilter, System.Diagnostics.TraceSource" },
-                { "sources:listeners:value:filter:value:level", "Critical" },
-            })
-            .Build();
+// There is an issue specific to NET48 that causes these tests not to pass.
+#if !NET48
```
Comment (eslutz) [inline RockLib.Diagnostics.Tests/ConfigurationExtensions/CreateDiagnosticsSettingsTests.cs:12]: After lots of debugging and then even more debugging with @JasonBock there is an issue specifically with .NET 4.8 causing these two tests to fail, so the tests will be skipped in that case.

File: RockLib.Diagnostics.Tests/DebugTraceListenerTests.cs [hunk 4]
```diff
@@ -5,20 +5,24 @@ namespace RockLib.Diagnostics.UnitTests
...
-        [Fact(DisplayName = "Constructor sets default name")]
-        public void ConstructorHappyPath1()
+        [Fact]
+        public void ConstructorSetsDefaultName()
         {
             var traceListener = new DebugTraceListener();
 
             traceListener.Name.Should().Be("");
+
+            traceListener.Dispose();
```
Comment (JasonBock) [inline RockLib.Diagnostics.Tests/DebugTraceListenerTests.cs]: `using var traceListener = new DebugTraceListener();` will call `Dispose()` for you. I'd change your code where you're calling `Dispose()` explicitly to this.

File: RockLib.Diagnostics.Tests/Tracing/ConfigureTraceTests.cs [hunk 5]
```diff
@@ -1,22 +1,24 @@
...
 {
+
     public class ConfigureTraceMethod
     {
         private static readonly bool InitialAutoFlush;
         private static readonly int InitialIndentSize;
         private static readonly bool InitialUseGlobalLock;
         private static readonly IReadOnlyList<TraceListener> InitialTraceListeners;
 
+#pragma warning disable CA1810 // Initialize reference type static fields inline
```
Comment (JasonBock) [inline RockLib.Diagnostics.Tests/Tracing/ConfigureTraceTests.cs]: I think you can set the fields inline, like:
```
private static readonly bool InitialAutoFlush = Trace.AutoFlush;
```
The `Initialize()` call still needs to be done in the type initializer.

**Review by JasonBock — CHANGES_REQUESTED** [review]

**Comment by kristycurrier** [comment]
What is the code coverage on this?

**Comment by eslutz** [comment]
> What is the code coverage on this?

@kristycurrier code coverage is at 96.86%

image

**Review by motionsuggests — APPROVED** [review]

**Comment by NeilSchiefer** [comment]
I think you need to update the appveyor config(s) right?

**Review by NeilSchiefer — APPROVED** [review]

**Pull request merged**